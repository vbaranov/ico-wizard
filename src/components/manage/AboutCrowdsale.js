import React from 'react'
import { Link } from 'react-router-dom'

export const AboutCrowdsale = ({ name, ticker, address, networkId }) => (
  <div className="about-step">
    <div className="step-icons step-icons_crowdsale-setup"/>
    <p className="title">{name} ({ticker}) Settings</p>
    <p className="description">
      The most important and exciting part of the crowdsale process. Here you can define parameters of your crowdsale
      campaign.
    </p>
    <Link to={`/crowdsale/?addr=${address}&networkID=${networkId}`} className="crowdsale-page-link">
      Crowdsale page
    </Link>
  </div>
)
