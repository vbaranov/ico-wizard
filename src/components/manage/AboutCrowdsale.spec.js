import React from 'react'
import { StaticRouter } from 'react-router'
import { AboutCrowdsale } from './AboutCrowdsale'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure } from 'enzyme'

configure({ adapter: new Adapter() })

describe('AboutCrowdsale', () => {
  it('should render the component', () => {
    const aboutCrowdsaleParams = {
      name: 'MyToken',
      ticker: 'MTK',
      address: '0x0000000000000000000000000000000000000001',
      networkId: '12648430',
    }

    expect(renderer.create(
      <StaticRouter location="testLocation" context={{}}>
        <AboutCrowdsale {...aboutCrowdsaleParams} />
      </StaticRouter>
    ).toJSON()).toMatchSnapshot()
  })
})
