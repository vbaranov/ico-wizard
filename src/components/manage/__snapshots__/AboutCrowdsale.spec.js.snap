// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutCrowdsale should render the component 1`] = `
<div
  className="about-step"
>
  <div
    className="step-icons step-icons_crowdsale-setup"
  />
  <p
    className="title"
  >
    MyToken
     (
    MTK
    ) Settings
  </p>
  <p
    className="description"
  >
    The most important and exciting part of the crowdsale process. Here you can define parameters of your crowdsale campaign.
  </p>
  <a
    className="crowdsale-page-link"
    href="/crowdsale/?addr=0x0000000000000000000000000000000000000001&networkID=12648430"
    onClick={[Function]}
  >
    Crowdsale page
  </a>
</div>
`;
